test/libsolidity/ASTRecoveryTest.h

```diff
@@ -0,0 +1,63 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <test/libsolidity/AnalysisFramework.h>
+#include <test/TestCase.h>
+#include <liblangutil/Exceptions.h>
+#include <libdevcore/AnsiColorized.h>
+
+#include <iosfwd>
+#include <string>
+#include <vector>
+#include <utility>
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+class ASTRecoveryTest: public TestCase
```

Comment:
Please document in a few words how the test is supposed to work.

Reply:
Will do and make the other changes in the next update.

Reply:
@erak looking at the corresponding sibling headers and `.cpp` files I can't find a single set of comment to follow on this. `ASTRecoverTest.{h,cpp}` are analogous to other "Interactive" tests.

I don't want to guess here at what format documentation you want since I will invariably get this wrong. Would you make a stab at documenting this, if there a blanks that need filling in, I can do that. Thanks.

Reply:
I think just something simple like "Validates that ASTs are generated even though the source files contain errors" is enough here, no need to follow a elaborate pattern for documentation here

Reply:
Thanks. Comment added in bd6d3c675